broadpeakplugin/src/main/java/com/kaltura/playkit/plugins/broadpeak/BroadpeakPlugin.java

```diff
@@ -240,13 +239,14 @@ public void apply(PKMediaEntry mediaEntry, PKMediaEntryInterceptor.Listener list
                 // Replace the URL
                 log.d("Apply New Entry URL  " + mediaEntry.getName() + " - " + mediaEntry.getId() + " url: " + result.getURL());
                 source.setUrl(result.getURL());
+                sessionsMap.put(source.getUrl(), session);
             } else {
                 // Stop the session in case of error
                 if (result != null) {
                     errorCode = result.getErrorCode();
                     errorMessage = result.getErrorMessage();
                 }
-                stopStreamingSession(source.getUrl());
+                session.stopStreamingSession();
```

Comment:
R we sure session will not be null? We previously had this protection cause we had issues

Reply:
The null check is couple lines above this code. So should never be null here